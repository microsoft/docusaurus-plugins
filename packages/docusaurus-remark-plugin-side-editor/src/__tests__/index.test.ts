import { describe, it, expect } from "vitest";
import path from "path";
import vfile from "to-vfile";
import mdx from "remark-mdx";
import remark from "remark";
import plugin from "../index";

const processFixture = async (name: string, options: any) => {
    const filePath = path.join(__dirname, "__fixtures__", `${name}.md`);
    const file = await vfile.read(filePath);
    const result = await remark().use(mdx).use(plugin, options).process(file);

    return result.toString();
};

const options = {
    languages: {
        ts: "typescript",
    },
};

describe("code-tabs plugin", () => {
    it("works on edit file", async () => {
        const result = await processFixture("edit", options);
        expect(result).toMatchSnapshot();
    });
});
