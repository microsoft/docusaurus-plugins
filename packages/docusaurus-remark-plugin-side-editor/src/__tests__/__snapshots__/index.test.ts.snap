// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code-tabs plugin works on edit file 1`] = `
"import SideEditorButton from '@theme/SideEditorButton';

Show edit button

\`\`\`js edit
console.log('edit this!')
\`\`\`

<SideEditorButton editorId={"js"} text={"console.log('edit this!')"} />

and that 

\`\`\`js edit=someeditor
console.log('edit this!')
\`\`\`

<SideEditorButton editorId={"someeditor"} text={"console.log('edit this!')"} />

\`\`\`ts edit
console.log('edit this!')
\`\`\`

<SideEditorButton editorId={"typescript"} text={"console.log('edit this!')"} />
"
`;
